effekt/shared/src/main/scala/effekt/generator/chez/Transformer.scala

```diff
@@ -117,9 +117,11 @@ trait Transformer {
     // currently bidirectional handlers are not supported
     case Resume(k, Return(expr)) => chez.Call(toChez(k), List(toChez(expr)))
 
+    case Resume(k, other) => sys error s"Not supported yet: ${util.show(stmt)}"
+
     case Region(body) => chez.Builtin("with-region", toChez(body))
 
-    case other => chez.Let(Nil, toChez(other))
+    case s: Scope => chez.Let(Nil, toChez(s))
```

Comment:
Annotated since otherwise we might be missing cases (which happened) and then loop during code gen.